### PR TITLE
Path error dialog

### PR DESCRIPTION
**Pull request by MDrakos**
https://www.pivotaltracker.com/story/show/174760146

### CODE REVIEW

File: installers/msi-language/ActiveState/Error.cs [hunk 1]
```diff
@@ -35,35 +34,57 @@ public static void ResetErrorDetails(Session session)
...
+    }
+
+    public class NetworkError
+    {
+        private const string type = "Network";
+
+        public static void SetDetails(Session session, string msg)
+        {
+            session.Log("Setting network error details");
+            Error.SetDetails(session, "Network", msg);
```
Comment (Naatan) [inline installers/msi-language/ActiveState/Error.cs:82]:
```suggestion
            Error.SetDetails(session, type, msg);
```

File: installers/msi-language/ActiveState/Error.cs [hunk 2]
```diff
@@ -35,35 +34,57 @@ public static void ResetErrorDetails(Session session)
...
+            Error.SetDetails(session, type, msg);
+        }
+
+        public static string Type()
+        {
+            return type;
+        }
+    }
+
+    public class NetworkError
```
Comment (Naatan) [inline installers/msi-language/ActiveState/Error.cs:75]: These 2 are basically identical aside from type. I'm sure C# has class inheritance that you could use here?

**Review by Naatan — CHANGES_REQUESTED** [review]
Please ensure you've tested all error cases before merging

File: installers/msi-language/ActiveState/Error.cs (hunk 2 above)
Comment (MDrakos) [reply installers/msi-language/ActiveState/Error.cs]: Yes we can. It changes how we use the errors but removes the duplicated code.

**Comment by MDrakos** [comment]
I've tested the 3 error dialogs and the regular install. It all works as expected.

**Review by Naatan — APPROVED** [review]

**Pull request merged**